app/helpers/automated_tests_helper.rb

```diff
@@ -404,7 +399,8 @@ def self.scripts_to_run(assignment, call_on)
 
   # Request an automated test. Ask Resque to enqueue a job.
   def self.async_test_request(grouping_id, call_on)
-    if files_available? && has_permission?
+    # when called on collection
+    if files_available? && (call_on == "collection" || has_permission?)

```

Comment:
Getting this error otherwise:
```
/home/vagrant/Markus/app/helpers/automated_tests_helper.rb:195:in `has_permission?': undefined method `admin?' for nil:NilClass (NoMethodError)
	from /home/vagrant/Markus/app/helpers/automated_tests_helper.rb:114:in `request_a_test_run'
```
which is caused from the call from submission_collector.rb

```
  if assignment.enable_test
      m_logger.log("Now requesting test run for #{assignment.short_identifier} \
                    on grouping: '#{grouping.id}'")
      AutomatedTestsHelper.request_a_test_run(new_submission.grouping.id,
                                              'collection',
                                              **@current_user**)
    end
```
Should we override the check, or otherwise what would be the prefered way to access/pass around @current_user?